File: group_anagrams.rs
use std::collections::HashMap;
struct Solution;

impl Solution {
    pub fn group_anagrams(strs: Vec<String>) -> Vec<Vec<String>> {
        let mut groups = HashMap::new();
        for s in strs {
            let hash = s.bytes().fold([0;26], |mut h, b| {
                h[(b - b'a') as usize] += 1; h
            });
            groups.entry(hash).or_insert_with(Vec::new).push(s);
        }
        groups.into_values().collect()
    }
}

File: other/group_anagrams.cpp
#include "leetcode.h"

class Solution {
public:
  vvd(string) groupAnagrams(vector<string> &strs) {
    map<array<int, 26>, vector<string>> groups;
    for (string s : strs) {
      array<int, 26> hash = {0};
      for (char c : s)
        ++hash[c - 'a'];
      groups[hash].push_back(s);
    }
    vvd(string) ans;
    for (auto [hash, group] : groups)
      ans.push_back(group);
    return ans;
  }
};
